.gitlab-ci.yml

```diff
@@ -1,14 +1,131 @@
+    - $PYTHON -m venv .env3
+    - source .env3/bin/activate
+    - pip install -r .virtualenv.dev-requirements.txt
+    - $PYTHON setup.py develop
+    - tox "-n $(nproc)"
+
+tox_37_check:
+  stage: test
+  image: opensuse/tumbleweed
+  variables:
+    TOXENV: 'unit_py3_7,check'
+    PYTHON: 'python3'
+  before_script:
+    - zypper --non-interactive refresh
+    - zypper --non-interactive install libxml2-devel libxslt-devel gcc python3 python3-devel python3-pip enchant ShellCheck which mkisofs xz gzip
+  <<: *tox_3_test
+
+tox_36_check:
+  stage: test
+  image: fedora:latest
+  variables:
+    TOXENV: 'unit_py3_6'
+    PYTHON: 'python3.6'
+  before_script:
+    - dnf install -y --refresh python36 tox libxml2-devel libxslt-devel gcc enchant make gcc which xz xorriso
+  <<: *tox_3_test
+
+tox_27_check:
+  stage: test
+  image: opensuse/tumbleweed
+  before_script:
+    - zypper --non-interactive refresh
+    - zypper --non-interactive install libxml2-devel libxslt-devel gcc python2 python2-devel python2-virtualenv python2-pip enchant which mkisofs xz gzip
+  script:
+    - virtualenv .env2
+    - source .env2/bin/activate
+    - pip install -r .virtualenv.dev-requirements.txt
+    - python2 setup.py develop
+    - tox -e unit_py2_7 "-n $(nproc)"
+
+OpenSUSE:
+  stage: test
+  image: opensuse/tumbleweed
+  before_script:
+    - zypper --non-interactive refresh
+    - zypper --non-interactive install python3 python3-devel python3-pip
+    # requirements
+    - zypper --non-interactive install python3-docopt python3-lxml python3-PyYAML python3-xattr python3-six python3-future python3-requests
+    # dev-requirements
+    - zypper --non-interactive install python3-pytest python3-pytest-cov python3-mock python3-flake8 python3-Sphinx python3-sphinx_rtd_theme python3-pyenchant python3-tox make gcc which mkisofs xz gzip
+    # pytest-xdist from the repos is broken, use the one from PyPI instead
+    - pip3 install sphinxcontrib-spelling pytest-xdist
+  <<: *distro_test
+
+Fedora:
+  stage: test
+  image: fedora:latest
+  <<: *fedora_install_deps
+  <<: *distro_test
+
+build_doc:
+  stage: doc
+  image: fedora:latest
+  <<: *fedora_install_deps
+  script:
+    - dnf -y install latexmk texlive-cmap texlive-metafont texlive-ec texlive-babel-english texlive-fncychap texlive-fancyhdr texlive-titlesec texlive-tabulary texlive-framed texlive-wrapfig texlive-parskip texlive-upquote texlive-capt-of texlive-needspace texlive-makeindex texlive-times texlive-helvetic texlive-courier texlive-gsftopk texlive-updmap-map texlive-dvips
+    - cd doc
+    - make html SPHINXBUILD=/usr/bin/sphinx-build-3
+    - make linkcheck SPHINXBUILD=/usr/bin/sphinx-build-3
+    - make man SPHINXBUILD=/usr/bin/sphinx-build-3
+    - make latexpdf SPHINXBUILD=/usr/bin/sphinx-build-3
+    # FIXME: there is a lot of warnings here, but they are not fatal
+    # => remove this maybe
+    - make spell SPHINXBUILD=/usr/bin/sphinx-build-3
+  artifacts:
+    paths:
+      - doc/build/
+
+RPM:
```

Comment:
I'm not sure if the RPM test makes sense. Since the packaging for the suse distro and also the packaging done by @Conan-Kudo is probably different compared to that test target ?

Reply:
It's not _that_ different. We _should_ make sure that it packages fine there anyway since you do build it in the OBS project for RH/Fedora distributions.

Reply:
My guess is that @Conan-Kudo uses very similar tools, but maybe patches the spec file a little.

Your call whether you want to have this: on one hand we get an extra safety that the spec file is functional before you publish on OBS, on the other hand it takes quite some time to build and it uses mock instead of obs-build (which I don't know how to include here).

Reply:
Aside from a single patch that changes xattr for pyxattr (which you can do with a targeted sed call), my packaging is more or less the same. The only other major change is that I ship the unversioned binaries in a kiwi-cli subpackage that `Provides: kiwi = %{version}-%{release}` so that OBS will always use this version instead of some legacy version.